### PR TITLE
Fix console.table display of Map and move tests to unit test

### PR DESCRIPTION
**Pull request by kevinkassimo**
The order of key and value in `forEach` callback on `Map` is reversed.

Also convert console integration tests into unit tests.

### CODE REVIEW

File: js/console_test.ts [hunk 1]
```diff
@@ -310,3 +318,272 @@ test(function consoleDetachedLog() {
...
+
+function mockConsole(f: ConsoleExamineFunc) {
+  const out = new StringBuffer();
+  const err = new StringBuffer();
+  const both = new StringBuffer();
+  const csl = new Console(
+    (x: string, isErr: boolean, printsNewLine: boolean) => {
+      const content = x + (printsNewLine ? "\n" : "");
+      const buf = isErr ? err : out;
+      buf.add(content);
```
Comment (rwaldron) [inline js/console_test.ts]: Hi! Sorry for the seemingly "drive-by" review question, but here goes: what is `buf` used for?

Comment (kevinkassimo) [reply js/console_test.ts]: For faking either stdout or stderr. I want to be able to view each log attempts in order through the chunks
